### PR TITLE
Add the possibility to import custom mdi icons

### PR DESCRIPTION
**Pull request by hututudor**
https://etvas.atlassian.net/browse/ETV-4232

### CODE REVIEW

File: src/Icon/Icon.jsx [hunk 1]
```diff
@@ -9,28 +9,51 @@ import animationSpeeds from '../assets/animationSpeeds'
...
 import sizes from '../assets/sizes'
 
-export const Icon = ({ name, size, color, rotate, ...props }) => {
-  const glyph = useMemo(() => {
+let currentGlyphs = { ...glyphs }
+
+export const addIcon = (name, icon) => {
+  if (currentGlyphs[name]) {
+    console.warn(`trying to redefine icon ${name}`)
+    return
```
Comment (alsofronie) [inline src/Icon/Icon.jsx:17]: Let the consumer redefine the icon, maybe this is the intended action:

```suggestion
    console.warn(`Warning: existing icon ${name} is overwritten`)
```

File: src/Icon/Icon.jsx [hunk 2]
```diff
@@ -9,28 +9,51 @@ import animationSpeeds from '../assets/animationSpeeds'
...
-    return name
+
+    return [name, false]
   }, [name])
 
-  if (typeof name === 'string' && typeof glyphs[name] === 'string') {
+  const path = useMemo(() => {
+    if (name) return currentGlyphs[name]
+    return face
+  }, [name, face])
```
Comment (alsofronie) [inline src/Icon/Icon.jsx:42]: Let face take precedence if the consumer specified the `face` prop:

```suggestion
  const path = useMemo(() => {
    if (face) {
      return face
    }
    return currentGlyphs[name]
  }, [name, face])
```

File: src/Icon/old-glyphs.json [hunk 3]
```diff
@@ -111,7 +111,6 @@
   "sticker": 59815,
   "starNo": 59813,
   "stack": 59812,
-  "search": 59792,
```
Comment (alsofronie) [inline src/Icon/old-glyphs.json:114]: Leave it until we drop the old icons altogether

Comment (hututudor) [reply src/Icon/old-glyphs.json:114]: If I leave this here, the mdi icon will be re-written by it

File: src/Icon/Icon.jsx [hunk 4]
```diff
@@ -9,28 +9,50 @@ import animationSpeeds from '../assets/animationSpeeds'
...
-    return name
+
+    return [name, false]
   }, [name])
 
-  if (typeof name === 'string' && typeof glyphs[name] === 'string') {
+  const path = useMemo(() => {
+    if (face) return face
+    return currentGlyphs[name]
+  }, [name, face])
```
Comment (alsofronie) [inline src/Icon/Icon.jsx:42]:
```suggestion
  const path = useMemo(() => face ? face : currentGlyphs[name], [name, face])
```

Comment (alsofronie) [inline src/Icon/Icon.jsx:42]:
```suggestion
```

File: src/Icon/Icon.jsx [hunk 5]
```diff
@@ -9,28 +9,50 @@ import animationSpeeds from '../assets/animationSpeeds'
...
+
+  if (!isOldVersion) {
     return (
       <BaseIcon
-        path={glyphs[name]}
+        path={path}
```
Comment (alsofronie) [inline src/Icon/Icon.jsx:47]:
```suggestion
        path={face ? face : currentGlyphs[name]}
```

File: src/Icon/old-glyphs.json (hunk 3 above)
Comment (alsofronie) [reply src/Icon/old-glyphs.json:114]: Make it so the new icons takes precedence over the old ones

**Review by alsofronie — CHANGES_REQUESTED** [review]

**Review by sergiuelisei — APPROVED** [review]

**Pull request merged**